Ability to over power TS and ignore specific error by developer

A developer should be able to add a comment above a ts error such as 

```
/// TS_IGNORE
let a:string = 1
```

and have the compiler not report that error...
there are certain scenarios the developer knows best and want to quiet down the error reports.

kind of like :any 

regards

Sean

Suppressing errors isn't necessarily about anti-patterns. I'm getting an incorrect error,

```
error TS1005: '{' expected.
```

on this perfectly fine JSX:

```jsx
<motor-node ref      = 'menu'
    absoluteSize     = `0, ${this.MENU_BAR_HEIGHT}, 0`
    >
    {menu}
</motor-node>,
```

It complains the the template string needs `{`. This should ideally be fixed, but until then, I would like to be able to suppress the error *for good reason*.

I am trying to manually set `window.console` for IE9 to prevent errors on `console.log` usage:

    if (!window.console)
        window.console = {};

But I get `error TS2540: Cannot assign to 'console' because it is a constant or a read-only property.` Are there any workaround for these use cases?

--- a/src/compiler/program.ts
+++ b/src/compiler/program.ts
@@ -1158,9 +1158,7 @@ namespace ts {
                 const programDiagnosticsInFile = programDiagnostics.getDiagnostics(sourceFile.fileName);
 
                 const diagnostics = bindDiagnostics.concat(checkDiagnostics, fileProcessingDiagnosticsInFile, programDiagnosticsInFile);
-                return isSourceFileJavaScript(sourceFile)
-                    ? filter(diagnostics, shouldReportDiagnostic)
-                    : diagnostics;
+                return filter(diagnostics, shouldReportDiagnostic);
             });
         }
 

